File: samples/src/main/java/tk/zielony/carbonsamples/MainActivity.java
package tk.zielony.carbonsamples;

import android.os.Bundle;

import carbon.component.ComponentItem;
import carbon.component.PaddingItem;

public class MainActivity extends SampleListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Samples.initToolbar(this, getString(R.string.app_name));

        setItems(new ComponentItem[]{
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf)),
                new SampleActivityItem(AnimationsActivity.class, getString(R.string.animationsActivity_title)),
                new SampleActivityItem(ColorsActivity.class, getString(R.string.colorsActivity_title)),
                new SampleActivityItem(ComparisonActivity.class, getString(R.string.comparisonActivity_title)),
                new SampleActivityItem(ComponentsActivity.class, getString(R.string.componentsActivity_title)),
                new SampleActivityItem(DemosActivity.class, getString(R.string.demosActivity_title)),
                new SampleActivityItem(DialogsActivity.class, getString(R.string.dialogsActivity_title)),
                new SampleActivityItem(FeaturesActivity.class, getString(R.string.featuresActivity_title)),
                new SampleActivityItem(GuidelinesActivity.class, getString(R.string.guidelinesActivity_title)),
                new SampleActivityItem(LibrariesActivity.class, getString(R.string.librariesActivity_title)),
                new SampleActivityItem(WidgetsActivity.class, getString(R.string.widgetsActivity_title)),
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf))
        });
    }

}

File: samples/src/main/java/tk/zielony/carbonsamples/MainActivity.kt
package tk.zielony.carbonsamples

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import carbon.recycler.ViewItemDecoration
import kotlinx.android.synthetic.main.activity_main.*

@SampleAnnotation(layoutId = R.layout.activity_main)
class MainActivity : SampleListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        about.setOnClickListener {
            startActivity(Intent(this, AboutActivity::class.java))
        }

        colors.setOnClickListener {
            startActivity(Intent(this, ColorsActivity::class.java))
        }

        val items = mutableListOf(
                "New and starred",
                SampleActivityGroup(NewSamplesActivity::class.java),
                SampleActivityGroup(FavouritesActivity::class.java),
                "All samples grouped by category",
                SampleActivityGroup(AnimationsActivity::class.java),
                SampleActivityGroup(ComponentsActivity::class.java),
                SampleActivityGroup(DemosActivity::class.java),
                SampleActivityGroup(WidgetsActivity::class.java),
                SampleActivityGroup(ChartsActivity::class.java),
                SampleActivityGroup(FeaturesActivity::class.java),
                SampleActivityGroup(GuidelinesActivity::class.java),
                SampleActivityGroup(LibrariesActivity::class.java),
                SampleActivityGroup(ThemesActivity::class.java),
                SampleActivityGroup(DialogsActivity::class.java)
        )

        val preferences = getSharedPreferences("samples", Context.MODE_PRIVATE)
        val recentlyUsed = preferences.getString(RECENTLY_USED, null)
        recentlyUsed?.let {
            items.addAll(0, listOf(
                    "The most recently used sample",
                    SampleActivityGroup(Class.forName(it) as Class<out Activity>)
            ))
        }

        val decoration = ViewItemDecoration(this, R.layout.carbon_row_divider)
        decoration.setDrawBefore { position -> position > 0 && items[position] is String }
        recycler.addItemDecoration(decoration)

        setItems(items)
    }

}
